service/history/workflow/cache/cache.go

```diff
@@ -227,6 +250,21 @@ func (c *CacheImpl) getOrCreateWorkflowExecutionInternal(
 		WorkflowKey: definition.NewWorkflowKey(namespaceID.String(), execution.GetWorkflowId(), execution.GetRunId()),
 		ShardUUID:   shardContext.GetOwner(),
 	}
+	c.mut.Lock()
+	c.demand[namespaceID.String()]++
+	handler.WithTags(metrics.NamespaceTag(namespaceID.String())).Gauge(metrics.CacheDemand.GetMetricName()).Record(float64(c.demand[namespaceID.String()]))
+	partition := namespaceID.String()
+	maxAllocation, ok := c.allocation[partition]
+	if !ok {
+		partition = unplannedCachePartition
+	}
+	// TODO what to do with unallocated partitions
+	if c.usage[partition] >= maxAllocation {
```

Comment:
You'll want this to be `c.usage[partition] + 1` since we're trying to add a request

Reply:
Yeah, that's right. Changed it.